src/eos-google-chrome-installer

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/python3
+#
+# eos-google-chrome-installer: helper script to install Google Chrome
+#
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Authors:
+#  Michal Rostecki <michal@kinvolk.io>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import argparse
+import logging
+import os
+import pathlib
+import subprocess
+import sys
+
+import dbus
+from dbus.mainloop import glib as dbus_glib
+import gi
+gi.require_version('Flatpak', '1.0')
+from gi.repository import Flatpak
+from gi.repository import Gio
+from gi.repository import GLib
+gi.require_version('NetworkManager', '1.0')
+from gi.repository import NetworkManager
+gi.require_version('NMClient', '1.0')
+from gi.repository import NMClient
+
+
+def exit_with_error(*args):
+    logging.error(*args)
+    sys.exit(1)
+
+
+class GoogleChromeInstaller:
+
+    GSETTINGS_SCHEMA = "com.endlessm.ChromeInitialSetup"
+
+    FLATPAK_CHROME_APP_ID = "com.google.Chrome"
+    FLATPAK_REMOTE_EOS_APPS = 'eos-apps'
+
+    def __init__(self):
+        if not self._check_gsetting():
+            exit_with_error("Google Chrome installation is disabled")
```

Comment:
As mentioned above, you can't exit here if called from the other script, since the gsetting could then prevent GS from opening even in the manual case when the user clicks explicitly on the icon when Chrome is not installed.

This gsetting should control the automatic behaviour only, not the manual one.